controls/SV-257811.rb

```diff
@@ -26,15 +26,60 @@
 $ sudo sysctl --system'
   impact 0.5
   ref 'DPMS Target Red Hat Enterprise Linux 9'
-  tag check_id: 'C-61552r925418_chk'
+  tag check_id: 'C-61552r942978_chk'
   tag severity: 'medium'
   tag gid: 'V-257811'
-  tag rid: 'SV-257811r925420_rule'
+  tag rid: 'SV-257811r942979_rule'
   tag stig_id: 'RHEL-09-213080'
   tag gtitle: 'SRG-OS-000132-GPOS-00067'
   tag fix_id: 'F-61476r925419_fix'
   tag satisfies: ['SRG-OS-000132-GPOS-00067', 'SRG-OS-000480-GPOS-00227']
   tag 'documentable'
   tag cci: ['CCI-000366', 'CCI-001082']
   tag nist: ['CM-6 b', 'SC-2']
+  tag 'host'
+
+  only_if('This system is acting as a router on the network, this control is Not Applicable', impact: 0.0) {
+    !input('network_router')
+  }
+
+  # Define the kernel parameter to be checked
+  parameter = 'kernel.yama.ptrace_scope'
+  action = 'usage of ptrace'
+  value = 1
+
+  # Get the current value of the kernel parameter
+  current_value = kernel_parameter(parameter)
+
+  # Check if the system is a Docker container
+  if virtualization.system.eql?('docker')
+    impact 0.0
+    describe 'Control not applicable within a container' do
+      skip 'Control not applicable within a container'
+    end
+  else
+
+    describe kernel_parameter(parameter) do
+      it 'is correctly set in the active kernel parameters' do
+        expect(current_value.value).to cmp value
+        expect(current_value.value).not_to be_nil
+      end
+    end
+
+    # Search for the kernel parameter in the configuration files
+    search_results = command("/usr/lib/systemd/systemd-sysctl --cat-config | egrep -v '^(#|;)' | grep -F #{parameter} | tail -1").stdout.strip
+
+    # Check the configuration files
+    describe 'Configuration files' do
+      if search_results.empty?
+        it "do not explicitly set the `#{parameter}` parameter" do
+          expect(search_results).not_to be_empty, "Add the line `#{parameter}=#{value}` to a file in the `/etc/sysctl.d/` directory"
+        end
+      else
+        it "set the parameter to the right value for #{action}" do
+          expect(search_results).to match(/#{parameter}\s*=\s*#{value}/)
```

Comment:
These regexes have also been highly inconsistent.  Here we are only looking for the parameter name then any whitespace then equal then any whitespace then the value.  In other places we don't allow for whitespace or we ensure that the value is at the end of the line.  IMO it should be closer to all of the above lol: `/^\s*#{parameter}\s*=\s*#{value}$/`